<?xml version="1.0" encoding="utf-8"?>
<?xml-stylesheet type="text/xsl" href="c:\program files (x86)\microsoft visual studio 14.0\team tools\static analysis tools\fxcop\Xml\CodeAnalysisReport.xsl"?>
<FxCopReport Version="14.0">
 <Targets>
  <Target Name="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\bin\OpenEnvironment.dll">
   <Modules>
    <Module Name="openenvironment.dll">
     <Namespaces>
      <Namespace Name="OpenEnvironment.Account">
       <Types>
        <Type Name="CustMembershipProvider" Kind="Class" Accessibility="Public" ExternallyVisible="True">
         <Members>
          <Member Name="#CreateUser(System.String,System.String,System.String,System.String,System.String,System.Boolean,System.Object,System.Web.Security.MembershipCreateStatus&amp;)" Kind="Method" Static="False" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\Account" File="CustomMembership.cs" Line="185">'CustMembershipProvider.CreateUser(string, string, string, string, string, bool, object, out MembershipCreateStatus)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
         </Members>
        </Type>
       </Types>
      </Namespace>
      <Namespace Name="OpenEnvironment.App_Logic.BusinessLogicLayer">
       <Types>
        <Type Name="Utils" Kind="Class" Accessibility="Assembly" ExternallyVisible="False">
         <Members>
          <Member Name="#RenderGridToExcelFormat(System.String,System.Web.UI.WebControls.GridView)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="False">
           <Messages>
            <Message TypeName="Do not dispose objects multiple times" Category="Microsoft.Usage" CheckId="CA2202" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="DependsOnFix">
             <Issue Certainty="75" Level="Warning" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\BusinessLogicLayer" File="Utils.cs" Line="451">Object 'sw' can be disposed more than once in method 'Utils.RenderGridToExcelFormat(string, GridView)'. To avoid generating a System.ObjectDisposedException you should not call Dispose more than one time on an object.: Lines: 451</Issue>
            </Message>
           </Messages>
          </Member>
         </Members>
        </Type>
       </Types>
      </Namespace>
      <Namespace Name="OpenEnvironment.App_Logic.DataAccessLayer">
       <Types>
        <Type Name="db_Accounts" Kind="Class" Accessibility="Public" ExternallyVisible="True">
         <Members>
          <Member Name="#GetT_OE_ROLESInUser(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Accounts.cs" Line="366">'db_Accounts.GetT_OE_ROLESInUser(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_OE_USERS()" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Accounts.cs" Line="53">'db_Accounts.GetT_OE_USERS()' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_OE_USERSByID(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Accounts.cs" Line="84">'db_Accounts.GetT_OE_USERSByID(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_OE_USERSByIDX(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Accounts.cs" Line="68">'db_Accounts.GetT_OE_USERSByIDX(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_OE_USERSInRole(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Accounts.cs" Line="319">'db_Accounts.GetT_OE_USERSInRole(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_OE_USERSNotInRole(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Accounts.cs" Line="346">'db_Accounts.GetT_OE_USERSNotInRole(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_VCCB_ROLEByIDX(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Accounts.cs" Line="248">'db_Accounts.GetT_VCCB_ROLEByIDX(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_VCCB_ROLEByName(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Accounts.cs" Line="233">'db_Accounts.GetT_VCCB_ROLEByName(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_VCCB_USERByEmail(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Accounts.cs" Line="99">'db_Accounts.GetT_VCCB_USERByEmail(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
         </Members>
        </Type>
        <Type Name="db_Ref" Kind="Class" Accessibility="Public" ExternallyVisible="True">
         <Members>
          <Member Name="#GetT_OE_APP_TASKS_ByName(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="49">'db_Ref.GetT_OE_APP_TASKS_ByName(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_ANAL_METHOD(System.Boolean)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="199">'db_Ref.GetT_WQX_REF_ANAL_METHOD(bool)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_ANAL_METHODByIDandContext(System.String,System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="269">'db_Ref.GetT_WQX_REF_ANAL_METHODByIDandContext(string, string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_ANAL_METHODByIDX(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="286">'db_Ref.GetT_WQX_REF_ANAL_METHODByIDX(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_CHAR_LIMITS_ByNameUnit(System.String,System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="791">'db_Ref.GetT_WQX_REF_CHAR_LIMITS_ByNameUnit(string, string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_CHAR_ORG(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="665">'db_Ref.GetT_WQX_REF_CHAR_ORG(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_CHAR_ORG_Count(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="700">'db_Ref.GetT_WQX_REF_CHAR_ORG_Count(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_CHAR_ORGByName(System.String,System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="683">'db_Ref.GetT_WQX_REF_CHAR_ORGByName(string, string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_CHARACTERISTIC(System.Boolean,System.Boolean)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="560">'db_Ref.GetT_WQX_REF_CHARACTERISTIC(bool, bool)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_CHARACTERISTIC_ByName(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="577">'db_Ref.GetT_WQX_REF_CHARACTERISTIC_ByName(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_CHARACTERISTIC_ByOrg(System.String,System.Boolean)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="645">'db_Ref.GetT_WQX_REF_CHARACTERISTIC_ByOrg(string, bool)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_COUNTY()" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="879">'db_Ref.GetT_WQX_REF_COUNTY()' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_DATA(System.String,System.Boolean,System.Boolean)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="376">'db_Ref.GetT_WQX_REF_DATA(string, bool, bool)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_DATA_ActivityTypeUsed(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="417">'db_Ref.GetT_WQX_REF_DATA_ActivityTypeUsed(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_DATA_Count()" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="495">'db_Ref.GetT_WQX_REF_DATA_Count()' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_DATA_TaxaSearch(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="397">'db_Ref.GetT_WQX_REF_DATA_TaxaSearch(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_DEFAULT_TIME_ZONE()" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="899">'db_Ref.GetT_WQX_REF_DEFAULT_TIME_ZONE()' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_DEFAULT_TIME_ZONE_ByName(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="916">'db_Ref.GetT_WQX_REF_DEFAULT_TIME_ZONE_ByName(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_LAB(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="937">'db_Ref.GetT_WQX_REF_LAB(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_LAB_ByIDandContext(System.String,System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="955">'db_Ref.GetT_WQX_REF_LAB_ByIDandContext(string, string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_SAMP_COL_METHOD_ByContext(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="1065">'db_Ref.GetT_WQX_REF_SAMP_COL_METHOD_ByContext(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_SAMP_COL_METHOD_ByIDandContext(System.String,System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="1048">'db_Ref.GetT_WQX_REF_SAMP_COL_METHOD_ByIDandContext(string, string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_SAMP_COL_METHOD_ByIDX(System.Nullable`1&lt;System.Int32&gt;)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="1030">'db_Ref.GetT_WQX_REF_SAMP_COL_METHOD_ByIDX(int?)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_SAMP_PREP_ByContext(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="1155">'db_Ref.GetT_WQX_REF_SAMP_PREP_ByContext(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_SAMP_PREP_ByIDandContext(System.String,System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="1138">'db_Ref.GetT_WQX_REF_SAMP_PREP_ByIDandContext(string, string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_TAXA_ByOrg(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="1332">'db_Ref.GetT_WQX_REF_TAXA_ByOrg(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_REF_TAXA_ORG(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="1252">'db_Ref.GetT_WQX_REF_TAXA_ORG(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetV_WQX_TRANSACTION_LOG(System.String,System.Nullable`1&lt;System.DateTime&gt;,System.Nullable`1&lt;System.DateTime&gt;,System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="176">'db_Ref.GetV_WQX_TRANSACTION_LOG(string, DateTime?, DateTime?, string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_IMPORT_LOG(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="1426">'db_Ref.GetWQX_IMPORT_LOG(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_IMPORT_LOG_NewActivity()" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="1444">'db_Ref.GetWQX_IMPORT_LOG_NewActivity()' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_TRANSACTION_LOG(System.String,System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="138">'db_Ref.GetWQX_TRANSACTION_LOG(string, int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_TRANSACTION_LOG_ByLogID(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_Ref.cs" Line="155">'db_Ref.GetWQX_TRANSACTION_LOG_ByLogID(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
         </Members>
        </Type>
        <Type Name="db_WQX" Kind="Class" Accessibility="Public" ExternallyVisible="True">
         <Members>
          <Member Name="#GetT_EPA_ORGS_ByOrgID(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1510">'db_WQX.GetT_EPA_ORGS_ByOrgID(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_OE_USERSInOrganization(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1701">'db_WQX.GetT_OE_USERSInOrganization(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_OE_USERSNotInOrganization(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1728">'db_WQX.GetT_OE_USERSNotInOrganization(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_OE_USERSPending(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1752">'db_WQX.GetT_OE_USERSPending(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_MONLOC_PendingInd(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="398">'db_WQX.GetT_WQX_MONLOC_PendingInd(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_RESULT(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1181">'db_WQX.GetT_WQX_RESULT(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_RESULT_ByIDX(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1198">'db_WQX.GetT_WQX_RESULT_ByIDX(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_RESULT_SampledCharacteristics(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1328">'db_WQX.GetT_WQX_RESULT_SampledCharacteristics(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetT_WQX_RESULTCount(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1217">'db_WQX.GetT_WQX_RESULTCount(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetV_WQX_ACTIVITY_LATEST(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="3740">'db_WQX.GetV_WQX_ACTIVITY_LATEST(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetV_WQX_ACTIVITY_LATESTByMonLocID(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="3757">'db_WQX.GetV_WQX_ACTIVITY_LATESTByMonLocID(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetV_WQX_ALL_ORGS()" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1547">'db_WQX.GetV_WQX_ALL_ORGS()' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetV_WQX_PENDING_RECORDS(System.String,System.Nullable`1&lt;System.DateTime&gt;,System.Nullable`1&lt;System.DateTime&gt;)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1569">'db_WQX.GetV_WQX_PENDING_RECORDS(string, DateTime?, DateTime?)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_ACTIVITY(System.Boolean,System.String,System.Nullable`1&lt;System.Int32&gt;,System.Nullable`1&lt;System.DateTime&gt;,System.Nullable`1&lt;System.DateTime&gt;,System.String,System.Boolean,System.Nullable`1&lt;System.Int32&gt;)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="834">'db_WQX.GetWQX_ACTIVITY(bool, string, int?, DateTime?, DateTime?, string, bool, int?)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_ACTIVITY_ByID(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="906">'db_WQX.GetWQX_ACTIVITY_ByID(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_ACTIVITY_ByUnique(System.String,System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="924">'db_WQX.GetWQX_ACTIVITY_ByUnique(string, string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_ACTIVITY_MyOrgCount(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1016">'db_WQX.GetWQX_ACTIVITY_MyOrgCount(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_ACTIVITYByMonLocID(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="942">'db_WQX.GetWQX_ACTIVITYByMonLocID(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_ACTIVITYByProjectID(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="960">'db_WQX.GetWQX_ACTIVITYByProjectID(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_ACTIVITYDisplay(System.Boolean,System.String,System.Nullable`1&lt;System.Int32&gt;,System.Nullable`1&lt;System.DateTime&gt;,System.Nullable`1&lt;System.DateTime&gt;,System.String,System.Boolean,System.Nullable`1&lt;System.Int32&gt;,System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="889">'db_WQX.GetWQX_ACTIVITYDisplay(bool, string, int?, DateTime?, DateTime?, string, bool, int?, string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_IMPORT_COL_ALIAS_byField(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="2100">'db_WQX.GetWQX_IMPORT_COL_ALIAS_byField(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_IMPORT_TEMP_MONLOC(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="2345">'db_WQX.GetWQX_IMPORT_TEMP_MONLOC(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_IMPORT_TEMP_MONLOC_ByID(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="2362">'db_WQX.GetWQX_IMPORT_TEMP_MONLOC_ByID(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_IMPORT_TEMP_PROJECT(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="2485">'db_WQX.GetWQX_IMPORT_TEMP_PROJECT(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_IMPORT_TEMP_PROJECT_ByID(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="2502">'db_WQX.GetWQX_IMPORT_TEMP_PROJECT_ByID(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_IMPORT_TEMP_RESULT_ByTempSampIDX(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="3559">'db_WQX.GetWQX_IMPORT_TEMP_RESULT_ByTempSampIDX(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_IMPORT_TEMP_SAMP_RESULT_Disp(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="3656">'db_WQX.GetWQX_IMPORT_TEMP_SAMP_RESULT_Disp(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_IMPORT_TEMP_SAMPLE_ByID(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="3031">'db_WQX.GetWQX_IMPORT_TEMP_SAMPLE_ByID(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_IMPORT_TEMP_SAMPLE_CountByUserID(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="3048">'db_WQX.GetWQX_IMPORT_TEMP_SAMPLE_CountByUserID(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_IMPORT_TEMPLATE(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1893">'db_WQX.GetWQX_IMPORT_TEMPLATE(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_IMPORT_TEMPLATE_DTL_byFieldMap(System.Int32,System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1933">'db_WQX.GetWQX_IMPORT_TEMPLATE_DTL_byFieldMap(int, string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_IMPORT_TEMPLATE_DTL_CharsByTemplateID(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1951">'db_WQX.GetWQX_IMPORT_TEMPLATE_DTL_CharsByTemplateID(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_IMPORT_TEMPLATE_DTL_DynamicByTemplateID(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1970">'db_WQX.GetWQX_IMPORT_TEMPLATE_DTL_DynamicByTemplateID(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_IMPORT_TEMPLATE_DTL_HardCodeByTemplateID(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1989">'db_WQX.GetWQX_IMPORT_TEMPLATE_DTL_HardCodeByTemplateID(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_MONLOC(System.Boolean,System.String,System.Nullable`1&lt;System.Boolean&gt;)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="303">'db_WQX.GetWQX_MONLOC(bool, string, bool?)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_MONLOC_ByID(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="345">'db_WQX.GetWQX_MONLOC_ByID(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_MONLOC_ByIDString(System.String,System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="380">'db_WQX.GetWQX_MONLOC_ByIDString(string, string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_MONLOC_ByImportID(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="362">'db_WQX.GetWQX_MONLOC_ByImportID(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_MONLOC_MyOrgCount(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="460">'db_WQX.GetWQX_MONLOC_MyOrgCount(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_ORGANIZATION()" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1349">'db_WQX.GetWQX_ORGANIZATION()' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_ORGANIZATION_ByID(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1384">'db_WQX.GetWQX_ORGANIZATION_ByID(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_ORGANIZATIONCanSubmit()" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1367">'db_WQX.GetWQX_ORGANIZATIONCanSubmit()' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_PROJECT(System.Boolean,System.String,System.Nullable`1&lt;System.Boolean&gt;)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="488">'db_WQX.GetWQX_PROJECT(bool, string, bool?)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_PROJECT_ByID(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="505">'db_WQX.GetWQX_PROJECT_ByID(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_PROJECT_ByIDString(System.String,System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="523">'db_WQX.GetWQX_PROJECT_ByIDString(string, string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_PROJECT_MyOrgCount(System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="642">'db_WQX.GetWQX_PROJECT_MyOrgCount(int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_USER_ORGS_AdminsByOrg(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1771">'db_WQX.GetWQX_USER_ORGS_AdminsByOrg(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_USER_ORGS_ByUserIDX(System.Int32,System.Boolean)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1592">'db_WQX.GetWQX_USER_ORGS_ByUserIDX(int, bool)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#GetWQX_USER_ORGS_ByUserIDX_OrgID(System.Int32,System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="1789">'db_WQX.GetWQX_USER_ORGS_ByUserIDX_OrgID(int, string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#SP_GenWQXXML_Org(System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="3705">'db_WQX.SP_GenWQXXML_Org(string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#SP_GenWQXXML_Single(System.String,System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="3675">'db_WQX.SP_GenWQXXML_Single(string, int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#SP_GenWQXXML_Single_Delete(System.String,System.Int32)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="3690">'db_WQX.SP_GenWQXXML_Single_Delete(string, int)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#SP_ImportActivityFromTemp(System.String,System.String,System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="3082">'db_WQX.SP_ImportActivityFromTemp(string, string, string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
          <Member Name="#SP_WQXAnalysis(System.String,System.String,System.Nullable`1&lt;System.Int32&gt;,System.String,System.Nullable`1&lt;System.DateTime&gt;,System.Nullable`1&lt;System.DateTime&gt;,System.String)" Kind="Method" Static="True" Accessibility="Public" ExternallyVisible="True">
           <Messages>
            <Message TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200" Status="Active" Created="2016-06-30 21:01:32Z" FixCategory="NonBreaking">
             <Issue Certainty="95" Level="Error" Path="C:\Users\dougtimms\Dropbox\open-environment\products\OpenWaters_git\open-waters\OpenWaters\OpenWaters\App_Logic\DataAccessLayer" File="db_WQX.cs" Line="3723">'db_WQX.SP_WQXAnalysis(string, string, int?, string, DateTime?, DateTime?, string)' rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Issue>
            </Message>
           </Messages>
          </Member>
         </Members>
        </Type>
       </Types>
      </Namespace>
     </Namespaces>
    </Module>
   </Modules>
  </Target>
 </Targets>
 <Rules>
  <Rule TypeName="Do not dispose objects multiple times" Category="Microsoft.Usage" CheckId="CA2202">
   <Name>Do not dispose objects multiple times</Name>
   <Description>A correctly implemented Dispose method can be called multiple times without throwing an exception. However, this is not guaranteed and to avoid generating a System.ObjectDisposedException you should not call Dispose more than one time on an object.</Description>
   <Resolution Name="Default">Object {0} can be disposed more than once in method {1}. To avoid generating a System.ObjectDisposedException you should not call Dispose more than one time on an object.: Lines: 451</Resolution>
   <Owner>RuleOwner</Owner>
   <Url>http://msdn.microsoft.com/library/ms182334.aspx</Url>
   <Email />
   <MessageLevel Certainty="75">Warning</MessageLevel>
   <File Name="dataflowrules.dll" Version="14.0.0.0" />
  </Rule>
  <Rule TypeName="RethrowToPreserveStackDetails" Category="Microsoft.Usage" CheckId="CA2200">
   <Name>Rethrow to preserve stack details</Name>
   <Description>Use the IL rethrow instruction to preserve original stack details when re-raising a caught exception.</Description>
   <Resolution Name="Default">{0} rethrows a caught exception and specifies it explicitly as an argument. Use 'throw' without an argument instead, in order to preserve the stack location where the exception was initially raised.</Resolution>
   <Owner />
   <Url>http://msdn.microsoft.com/library/ms182363.aspx</Url>
   <Email>[none]</Email>
   <MessageLevel Certainty="95">Error</MessageLevel>
   <File Name="usagerules.dll" Version="14.0.0.0" />
  </Rule>
 </Rules>
 <Localized>
  <String Key="Category">Category</String>
  <String Key="Certainty">Certainty</String>
  <String Key="CollapseAll">Collapse All</String>
  <String Key="CheckId">Check Id</String>
  <String Key="Error">Error</String>
  <String Key="Errors">error(s)</String>
  <String Key="ExpandAll">Expand All</String>
  <String Key="Help">Help</String>
  <String Key="Line">Line</String>
  <String Key="Messages">message(s)</String>
  <String Key="LocationNotStoredInPdb">[Location not stored in Pdb]</String>
  <String Key="Project">Project</String>
  <String Key="Resolution">Resolution</String>
  <String Key="Rule">Rule</String>
  <String Key="RuleFile">Rule File</String>
  <String Key="RuleDescription">Rule Description</String>
  <String Key="Source">Source</String>
  <String Key="Status">Status</String>
  <String Key="Target">Target</String>
  <String Key="Warning">Warning</String>
  <String Key="Warnings">warning(s)</String>
  <String Key="ReportTitle">Code Analysis Report</String>
 </Localized>
</FxCopReport>
